### PR TITLE
8302149: Speed up compiler/jsr292/methodHandleExceptions/TestAMEnotNPE.java

### PR DESCRIPTION
**Pull request by eme64**
Speed up via compile restriction.
For the 3 jobs, time used to be:
` 0.7`, `1.5`, `25.7`
Now the third one `-Xcomp` is much faster:
`0.7`, `1.5`, `2.0`

To quote @iwanowww :
```
According to its name, the test ensures exceptions are thrown in proper order (AME precedes NPE).
So, it’s enough to compile the methods which contain such call sites.
I believe they reside in methods named “test”.
```

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8302149](https://bugs.openjdk.org/browse/JDK-8302149): Speed up compiler/jsr292/methodHandleExceptions/TestAMEnotNPE.java

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)
 * [Vladimir Ivanov](https://openjdk.org/census#vlivanov) (@iwanowww - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12536/head:pull/12536` \
`$ git checkout pull/12536`

Update a local copy of the PR: \
`$ git checkout pull/12536` \
`$ git pull https://git.openjdk.org/jdk pull/12536/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12536`

View PR using the GUI difftool: \
`$ git pr show -t 12536`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12536.diff">https://git.openjdk.org/jdk/pull/12536.diff</a>

</details>

### CODE REVIEW

**Review by vnkozlov — APPROVED** [review]
Good.

File: test/hotspot/jtreg/compiler/jsr292/methodHandleExceptions/TestAMEnotNPE.java [hunk 1]
```diff
@@ -30,8 +30,11 @@
  *
  * @build p.*
  * @run main/othervm compiler.jsr292.methodHandleExceptions.TestAMEnotNPE
- * @run main/othervm -Xint compiler.jsr292.methodHandleExceptions.TestAMEnotNPE
- * @run main/othervm -Xcomp compiler.jsr292.methodHandleExceptions.TestAMEnotNPE
+ * @run main/othervm -Xint
+ *                   compiler.jsr292.methodHandleExceptions.TestAMEnotNPE
+ * @run main/othervm -Xcomp
+ *                   -XX:CompileCommand=compileonly,*::test
```
Comment (vnkozlov) [inline test/hotspot/jtreg/compiler/jsr292/methodHandleExceptions/TestAMEnotNPE.java:36]: What time you get if you specify all methods in test's class instead:
`compiler.jsr292.methodHandleExceptions.TestAMEnotNPE::*`

Comment (eme64) [reply test/hotspot/jtreg/compiler/jsr292/methodHandleExceptions/TestAMEnotNPE.java:36]: It takes just a tiny bit longer. `2.3` seconds for that run, whereas with `-XX:CompileCommand=compileonly,*::test` I get `2` seconds.

Comment (vnkozlov) [reply test/hotspot/jtreg/compiler/jsr292/methodHandleExceptions/TestAMEnotNPE.java:36]: I suggest then to use my proposal - compile all test's methods. I concern that compiling only one method may not catch some issues when the method is inlined.

Comment (eme64) [reply test/hotspot/jtreg/compiler/jsr292/methodHandleExceptions/TestAMEnotNPE.java:36]: Actually, I just checked what methids it wants to compile. And the generated `test` methods are these:
```
p.Tdirect::test
p.T::test
p.Treflect::test
```

There is also a method `p.I::m` that seems to have been generated. Does that also need to be compiled?
I also see a `p.C::<init>` and `p.E::<init>.` And some class or interface `p.D`. And `p.F::<init>` and `p.F::m`.

@iwanowww do you think we need to include all of these to be compiled?

Comment (vnkozlov) [reply test/hotspot/jtreg/compiler/jsr292/methodHandleExceptions/TestAMEnotNPE.java:36]: I think we can have several `compileonly` commands for different methods so you can list them all. Please, verify that.

Comment (iwanowww) [reply test/hotspot/jtreg/compiler/jsr292/methodHandleExceptions/TestAMEnotNPE.java:36]: I agree that it's easier to reason about the test when all generated methods are compiled.  I see it put generated classes in packages `p` and `q`. Compiling everything from those packages sounds fine.

**Review by vnkozlov — APPROVED** [review]
Update is good.

**Review by iwanowww — APPROVED** [review]
Looks good.

**Comment by eme64** [comment]
Thanks @iwanowww @vnkozlov for the suggestions and reviews!
/integrate

**Pull request closed without merging**